src/librustc/infer/error_reporting/anon_anon_conflict.rs

```diff
@@ -176,8 +234,8 @@ impl<'a, 'gcx, 'tcx> Visitor<'gcx> for FindNestedTypeVisitor<'a, 'gcx, 'tcx> {
             hir::TyRptr(ref lifetime, _) => {
                 match self.infcx.tcx.named_region_map.defs.get(&lifetime.id) {
                     // the lifetime of the TyRptr
-                    Some(&rl::Region::LateBoundAnon(debuijn_index, anon_index)) => {
-                        if debuijn_index.depth == 1 && anon_index == br_index {
+                    Some(&rl::Region::LateBoundAnon(debruijn_index, anon_index)) => {
+                        if debruijn_index.depth == 1 && anon_index == br_index {
```

Comment:
This might be hard to fix, and it will only cause silly error messages (rather than e.g. crashing the compiler, which other debruijn index mistakes did), so I think it's ok if you leave it broken for now.